stdlib/Michelson.ju

```diff
@@ -0,0 +1,143 @@
+  let map       = %Michelson.map
+  let big-map   = %Michelson.big-map
+
+  -- primitive types
+  let key       = %Michelson.key
+  let unit      = %Michelson.unit
+  let signature = %Michelson.signature
+  let chain-id  = %Michelson.chain-id
+  let int       = %Michelson.int
+  let nat       = %Michelson.nat
+  let string    = %Michelson.string
+  let bytes     = %Michelson.bytes
+  let mutez     = %Michelson.mutez
+  let bool      = %Michelson.bool
+  let hash-key  = %Michelson.hash-key
+  let timestamp = %Michelson.timestamp
+  let address   = %Michelson.address
+
+  -- arithmetic functions
+  sig add : (x: type) -> x -> x -> x
+  let add = %Michelson.add
+
+  sig mul : (x: type) -> x -> x -> x
+  let mul = %Michelson.mul
+
+  sig sub : (x: type) -> x -> x -> x
+  let sub = %Michelson.sub
+
+  sig div : (x: type) -> x -> x -> x
+  let div = %Michelson.div
+
+  sig abs : (x: type) -> x -> x -> x
+  let abs = %Michelson.abs
+
+  -- boolean functions
+  sig and' : (x: type) -> x -> x -> x
+  let and' = %Michelson.and
+
+  sig or  : (x: type) -> x -> x -> x
+  let or = %Michelson.or
+
+  sig xor : (x: type) -> x -> x -> x
+  let xor = %Michelson.xor
+
+  sig not : (x: type) -> x -> x -> x
+  let not = %Michelson.not
+
+  -- other functions
+  sig () : unit
+  let () = %Michelson.unit
+
+  sig now : timestamp
+  let now = %Michelson.now
+
+  sig amount : (x : let) -> x -> ()
+  let amount = %Michelson.amount
+
+  sig balance : (x : let) -> address -> x
+  let balance = %Michelson.balance
+
+  sig sha256 : (x: type) -> x -> hash-key
+  let sha256 = %Michelson.sha256
+
+  sig sha512 :
+  let sha512 = %Michelson.sha512
+
+  sig blake2b :
+  let blake2b = %Michelson.
+
+  sig source : address
+  let source = %Michelson.source
+
+  sig sender : address
+  let sender = %Michelson.sender
+
+  sig implicit-account : address
+  let implicit-account = %Michelson.implicit-account
+
+  sig create-contract : () -> signature
+  let create-contract = %Michelson.create-contract
+
+  sig transfer-tokens : (x: let) -> address -> x -> hash-key
+  let transfer-token = %Michelson.transfer-tokens
+
+  sig set-delegate : address -> ()
+  let set-delegate = %Michelson.set-delegate
+
+  sig none : either () a
+  let none = %Michelson.none
+
+  sig some : b -> either a b
+  let some = %Michelson.some
+
+  sig left : a -> either a b
+  let left = %Michelson.left
+
+  sig right : b -> either a b
+  let right = %Michelson.right
+
+  sig nil : list a
+  let nil = %Michelson.nil
+
+  sig cons : a -> list a
+  let cons = %Michelson.cons
+
+  sig car : list a -> a
+  let car = %Michelson.car
+
+  sig cdr : list a -> list a
+  let cdr = %Michelson.cdr
+
+  sig size : list a -> int
+  let size = %Michelson.size
+
+  sig empty-set : set a
+  let empty-set = %Michelson.empty-set
+
+  sig empty-map : map a
+  let empty-map = %Michelson.empty-map
+
+  sig empty-big-map : big-map a
+  let empty-big-map = %Michelson.empty-big-map
+
+  sig map : (x: type) -> lambda a b -> x a -> x b
```

Comment:
The compiler could do the `lambda` wrapping automatically but maybe explicit is better (like you have here)

Reply:
ahh I missed thing, `(a -> b)` works for lambda